raiden/network/proxies/token_network.py

```diff
@@ -0,0 +1,773 @@
+            **participants_data,
+        }
+
+    def locked_amount_by_locksroot(
+            self,
+            participant: typing.Address,
+            partner: typing.Address,
+            locksroot: typing.Locksroot,
+    ) -> int:
+        """ Returns the locked amount for a specific participant's locksroot.
+        """
+        data = self._call_and_check_result(
+            'getParticipantLockedAmount',
+            participant,
+            partner,
+            locksroot
+        )
+        return data
+
+    def settle_block_number(self, partner: typing.Address):
+        """ Returns the channel settle_block_number. """
+        channel_data = self.detail_channel(partner)
+        return channel_data.get('settle_block_number')
+
+    def channel_is_opened(self, partner: typing.Address):
+        """ Returns true if the channel is in a closed state, false otherwise. """
+        channel_data = self.detail_channel(partner)
+
+        if not isinstance(channel_data['state'], typing.T_ChannelState):
+            raise ValueError('channel state must be of type ChannelState')
+
+        return channel_data.get('state') == CHANNEL_STATE_OPENED
+
+    def channel_is_closed(self, partner: typing.Address):
+        """ Returns true if the channel is in a closed state, false otherwise. """
+        channel_data = self.detail_channel(partner)
+
+        if not isinstance(channel_data['state'], typing.T_ChannelState):
+            raise ValueError('channel state must be of type ChannelState')
+
+        return channel_data.get('state') == CHANNEL_STATE_CLOSED
+
+    def channel_is_settled(self, partner: typing.Address):
+        """ Returns true if the channel is in a closed state, false otherwise. """
+        channel_data = self.detail_channel(partner)
+
+        if not isinstance(channel_data['state'], typing.T_ChannelState):
+            raise ValueError('channel state must be of type ChannelState')
+
+        return channel_data.get('state') == CHANNEL_STATE_SETTLED
+
+    def closing_address(self, partner: typing.Address):
+        """ Returns the address of the closer, if the channel is closed, None
+        otherwise. """
+        participants_data = self.detail_participants(partner)
+
+        if participants_data.get('our_is_closer'):
+            return participants_data.get('our_address')
+        elif participants_data.get('partner_is_closer'):
+            return participants_data.get('partner_address')
+
+        return None
+
+    def can_transfer(self, partner: typing.Address):
+        """ Returns True if the channel is opened and the node has deposit in
+        it.
+
+        Note: Having a deposit does not imply having a balance for off-chain
+        transfers. """
+        closed = self.channel_is_closed(partner)
+
+        if closed is True:
+            return False
+
+        return self._detail_participant(self.node_address, partner)['deposit'] > 0
+
+    def deposit(self, total_deposit, partner: typing.Address):
```

Comment:
since you already added typing to this function please also add a type for `total_deposit`. I guess it should be an int.